src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/UnconfiguredProjectHostObject.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
+{
+    /// <summary>
+    /// Language service host object for an unconfigured project that wraps the underlying host object.
+    /// </summary>
+    internal sealed class UnconfiguredProjectHostObject : AbstractHostObject, IUnconfiguredProjectHostObject
+    {
+        private readonly IVsHierarchy _innerHierarchy;
+        private readonly Dictionary<uint, IVsHierarchyEvents> _hierEventSinks;
+
+        public UnconfiguredProjectHostObject(UnconfiguredProject unconfiguredProject)
+        {
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+
+            _innerHierarchy = unconfiguredProject.Services.HostObject as IVsHierarchy;
+            _hierEventSinks = new Dictionary<uint, IVsHierarchyEvents>();
+        }
+
+        protected override IVsHierarchy InnerHierarchy => _innerHierarchy;
+        public IConfiguredProjectHostObject ActiveIntellisenseProjectHostObject { get; set; }
+        public override String ActiveIntellisenseProjectDisplayName => ActiveIntellisenseProjectHostObject?.ProjectDisplayName;
+
+        #region IVsHierarchy overrides
+
+        public override int AdviseHierarchyEvents(IVsHierarchyEvents pEventSink, out uint pdwCookie)
+        {
+            var hr = base.AdviseHierarchyEvents(pEventSink, out pdwCookie);
+            if (hr == VSConstants.S_OK)
+            {
+                _hierEventSinks.Add(pdwCookie, pEventSink);
+            }
+
+            return hr;
+        }
+
+        public override int GetProperty(uint itemid, int propid, out Object pvar)
+        {
+            if (ActiveIntellisenseProjectHostObject != null)
+            {
+                switch (propid)
+                {
+                    case (int)__VSHPROPID8.VSHPROPID_ActiveIntellisenseProjectContext:
+                        pvar = ActiveIntellisenseProjectDisplayName;
```

Comment:
This is indeed the display name for the active intellisense context you set via the navigation bar. Each configured project context has a display name, and the active context project's name is tracked as ActiveIntellisenseProjectDisplayName by the unconfigured project host object.